docs/SNAPD_CONFIGURATION.md

```diff
@@ -0,0 +1,212 @@
+# Gomaxprocs sets the number of cores to use on the system
+# for snapd to use. Default for gomaxprocs is 1
+gomaxprocs: 1
+```
+
+### snapd control configurations
+The control section contains settings for configuring the Control module within the snap daemon. These configuration settings are specific for the running of plugins and the plugins section under control allows for passing of plugin specific configuration items to the plugins during a task run.
+
+```yaml
+control:
+  # auto_discover_path sets a directory to auto load plugins on the start
+  # of the snap daemon
+  auto_discover_path: /opt/snap/plugins
+
+  # cache_expiration sets the time interval for the plugin cache to use before
+  # expiring collection results from collect plugins. Default value is 500ms
+  cache_expiration: 500ms
+
+  # max_running_plugins sets the size of the available plugin pool for each
+  # plugin loaded in the system. Default value is 3
+  max_running_plugins: 3
+
+  # keyring_paths sets the directory(s) to search for keyring files for signed
+  # plugins. This can be a comma separated list of directories
+  keyring_paths: /opt/snap/plugins/keyrings
+
+  # plugin_trust_level sets the plugin trust level for snapd. The default state
+  # for plugin trust level is enabled (1). When enabled, only signed plugins that can
+  # be verified will be loaded into snapd. Signatures are verifed from
+  # keyring files specided in keyring_path. Plugin trust can be disabled (0) which
+  # will allow loading of all plugins whether signed or not. The warning state allows
+  # for loading of signed and unsigned plugins. Warning messages will be displayed if
+  # an unsigned plugin is loaded. Any signed plugins that can not be verified will
+  # not be loaded. Valid values are 0 - Off, 1 - Enabled, 2 - Warning
+  plugin_trust_level: 1
+  
+  # plugins section contains plugin config settings that will be applied for
+  # plugins across tasks.
+  plugins:
+    all:
+      password: p@ssw0rd
+    collector:
+      all:
+        user: jane
+      pcm:
+        all:
+          path: /usr/local/pcm/bin
+        versions:
+          1:
+            user: john
+            someint: 1234
+            somefloat: 3.14
+            somebool: true
+      psutil:
+        all:
+          path: /usr/local/bin/psutil
+    publisher:
+      influxdb:
+        all:
+          server: xyz.local
+          password: $password
+    processor:
+      movingaverage:
+        all:
+          user: jane
+        versions:
+          1:
+            user: tiffany
+            password: new password
+```
+
+### snapd scheduler configurations
+The scheduler section of the configuration file configures settings for the Scheduler module inside the snap daemon.
+
+```yaml
+scheduler:
+  # work_manager_queue_size sets the size of the worker queue inside snapd scheduler.
+  # Default value is 25.
+  work_manager_queue_size: 25
+
+  # work_manager_pool_size sets the size of the <blank>. Default value is 4.
```

Comment:
The line `# work_manager_pool_size sets the size of the <blank>. Default value is 4.` I wasn't sure: 

* what `<blank>` was supposed to be
* how much description is worth adding? 

@tjmcs added a little context that does the trick!